packages/expo-gl/src/Canvas.tsx

```diff
@@ -37,7 +37,18 @@ const CanvasWrapper: React.FunctionComponent<
   const ref = React.useRef<View>(null);
   const _canvasRef = React.useRef<HTMLCanvasElement>(null);
 
-  function updateCanvasSize(): void {
+  const getSize = React.useCallback((): { width: number; height: number } => {
+    if (size) {
+      return size;
+    } else if (!ref.current || typeof window === 'undefined') {
+      return { width: 0, height: 0 };
+    }
+    const element = getElement(ref.current);
+    const { offsetWidth: width = 0, offsetHeight: height = 0 } = element;
+    return { width, height };
+  }, [!!size, ref.current])
```

Comment:
Was there any reason for converting the size to a boolean value?

Reply:
Hello! I had a thought about the `size` object. Since JavaScript compares objects by reference, converting it to a Boolean for comparison might be more stable and performant, especially since the function seems to only check for the presence of the `size` object, not its specific values. Alternatively, we could directly compare `size.width` and `size.height` if we need to examine the dimensions.

I apologize if my understanding is incorrect.

Reply:
> Was there any reason for converting the size to a boolean value?

https://github.com/expo/expo/pull/35601#issuecomment-2800139074